File: prince-of-versions/src/main/java/co/infinum/princeofversions/StreamIo.java
package co.infinum.princeofversions;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.nio.charset.Charset;

/**
 * Utility class for reading from input stream into string.
 */
class StreamIo {

    private static final Charset DEFAULT_CHARSET = Charset.forName("UTF-8");

    private StreamIo() {

    }

    /**
     * Transform input stream into string using UTF-8 charset.
     *
     * @param is Input stream.
     * @return String read from stream.
     * @throws IOException if reading error occurred.
     */
    static String toString(InputStream is) throws IOException {
        return toString(is, DEFAULT_CHARSET);
    }

    /**
     * Transforming input stream into string using given charset.
     *
     * @param is      Input stream.
     * @param charset Charset used while reading stream.
     * @return String read from stream.
     * @throws IOException if reading error occurred.
     */
    static String toString(InputStream is, Charset charset) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(is, charset));
        StringBuilder out = new StringBuilder();
        String line;
        while ((line = reader.readLine()) != null) {
            out.append(line);
        }
        try {
            reader.close();
        } catch (Exception ignorable) { // NOPMD
            // ignorable exception
        }
        return out.toString();
    }
}

File: prince-of-versions/src/main/java/co/infinum/princeofversions/StreamIo.kt
package co.infinum.princeofversions

import java.io.IOException
import java.io.InputStream
import java.nio.charset.Charset

/**
 * Utility object for reading from an InputStream into a String.
 */
internal object StreamIo {

    private val DEFAULT_CHARSET: Charset = Charsets.UTF_8

    /**
     * Transforms an InputStream into a String, concatenating all lines.
     * Any exception thrown while closing the stream will be ignored.
     *
     * @param inputStream The InputStream to read from.
     * @param charset The charset to use for decoding. Defaults to UTF-8.
     * @return The String read from the stream.
     * @throws IOException if a reading error occurs.
     */
    @JvmStatic
    @JvmOverloads
    @Throws(IOException::class)
    fun toString(inputStream: InputStream, charset: Charset = DEFAULT_CHARSET): String {
        val reader = inputStream.bufferedReader(charset)
        val out = StringBuilder()
        try {
            var line = reader.readLine()
            while (line != null) {
                out.append(line)
                line = reader.readLine()
            }
        } finally {
            try {
                reader.close()
            } catch (ignorable: IOException) {
                // This exception is intentionally ignored to match the original Java implementation.
            }
        }
        return out.toString()
    }
}
